contractcourt/chain_arbitrator.go

```diff
@@ -509,44 +517,24 @@ func (c *ChainArbitrator) ResolveContract(chanPoint wire.OutPoint) error {
 		return err
 	}
 
-	// Now that the channel has been marked as fully closed, we'll stop
```

Comment:
So what's he deadlock scenario here? That the channel arb calls this function while the chain arb is trying to stop it? 

I think that can alternatively be handled with an async call from the chan arb. At that point, it's shutting down, and can't really do much with any error returned here as all the contracts have been resolved (channel is fully closed).